packages/peregrine/lib/talons/CartPage/GiftCards/useApplyButton.js

```diff
@@ -0,0 +1,18 @@
+import { useCallback } from 'react';
+import { useFormState } from 'informed';
+
+export const useApplyButton = props => {
+    const { handleApplyCard } = props;
+
+    const giftCardEntryFormState = useFormState();
+
+    const handleApplyCardWithCode = useCallback(() => {
+        const giftCardCode = giftCardEntryFormState.values['card'];
```

Comment:
The `ApplyButton` and `CheckBalanceButton` are child components / separated out so that they can `useFormState` to get this card value from the `Form`.

This is different from Coupons, which could use the `Form`'s `onSubmit` handler directly. Since gift cards has multiple action buttons (apply and check balance), it needed to do it this way.

I suppose we could make one of them (Apply, I'd say) a submit button and only do this for `CheckBalanceButton`.

Reply:
I discussed this with @sirugh last week, but I think multiple submit is the way to go here. From the docs:

> **`submitForm`**
> 
> This function will submit the form and trigger some lifecycle events.
> 1. It will set all the fields to touched.
> 2. It will call all validators.
> 3. It will call `onSubmit` if the form is valid. This function can be called manually however it is also called if you have a `<button type='submit'>` within the `<Form>.`

As you can see, form submission involves more than just calling an endpoint with some values. And we certainly don't want to be in the business of replicating these effects ourselves. So we have two options here:

- Let the buttons trigger a `submit` event on the form as normal, or
- Get a handle on the form and call the `submitForm()` method

The calls should be done inside of a `handleSubmit` callback passed to the `Form` element via `onSubmit`. The first argument to that handler is `values`, so it should be very easy to get the value and make the call. You should also have access to event (and from it, the button element that triggered the event), which you can use to determine which call to make. Let me know if you have difficulty with this part.

The official attribute for distinguishing between multiple submit buttons is [`formaction`](https://css-tricks.com/separate-form-submit-buttons-go-different-urls/), I believe.

Reply:
> You should also have access to event (and from it, the button element that triggered the event), which you can use to determine which call to make. Let me know if you have difficulty with this part.

When I tried it previously I definitely was not getting this value but I'll take another crack at it.

Reply:
Ok, the callback passed to `Form` `onSubmit` only gets the values of the `Form`, not the `event`: https://github.com/joepuzzo/informed/blob/d797b3d5a51f2125039cd4c315429097b41aa14a/src/hooks/useForm.js#L51

But I was able to get this working: https://github.com/magento/pwa-studio/pull/2124/commits/41da0823a2f82141c2ff27f8dd485c62ea596d89.

Basically since I was already keeping track of the "last action taken" (apply, check balance, remove, toggle), I was able to piggyback on that and combine it with the `values` passed to `onSubmit` to make the appropriate call.

It muddles the talon a bit, but it lets me remove child components and keep everything centralized.